Makefile

```
@@ -15,101 +15,83 @@
-MACHINE_EMULATOR_TOOLS_DEB     := machine-emulator-tools-$(MACHINE_EMULATOR_TOOLS_VERSION).deb
-MACHINE_EMULATOR_TOOLS_IMAGE   := cartesi/machine-emulator-tools:$(MACHINE_EMULATOR_TOOLS_VERSION)
-
-LINUX_SOURCES_VERSION  ?= 5.15.63-ctsi-2
-LINUX_SOURCES_FILEPATH := dep/linux-$(LINUX_SOURCES_VERSION).tar.gz
-LINUX_SOURCES_URLPATH  := https://github.com/cartesi/linux/archive/refs/tags/v$(LINUX_SOURCES_VERSION).tar.gz
-
-RNDADDENTROPY_VERSION  ?= 3.0.0
-RNDADDENTROPY_FILEPATH := dep/twuewand-$(RNDADDENTROPY_VERSION).tar.gz
-RNDADDENTROPY_URLPATH  := https://www.finnie.org/software/twuewand/twuewand-$(RNDADDENTROPY_VERSION).tar.gz
-
-SHASUMFILES := $(LINUX_SOURCES_FILEPATH) $(RNDADDENTROPY_FILEPATH)
-
-all: build copy
-
-build: Dockerfile checksum
-	docker buildx build --platform=linux/riscv64 --load \
-		--build-arg MACHINE_EMULATOR_TOOLS_TAR_GZ=$(MACHINE_EMULATOR_TOOLS_TAR_GZ) \
-		--build-arg MACHINE_EMULATOR_TOOLS_DEB=$(MACHINE_EMULATOR_TOOLS_DEB) \
-		--build-arg LINUX_SOURCES_VERSION=$(LINUX_SOURCES_VERSION) \
-		--build-arg LINUX_SOURCES_FILEPATH=$(LINUX_SOURCES_FILEPATH) \
-		--build-arg RNDADDENTROPY_VERSION=$(RNDADDENTROPY_VERSION) \
-		--build-arg RNDADDENTROPY_FILEPATH=$(RNDADDENTROPY_FILEPATH) \
-		-t $(MACHINE_EMULATOR_TOOLS_IMAGE) \
-		-f $< \
-		.
+TOOLS_DEB     := machine-emulator-tools-v$(VERSION).deb
+TOOLS_IMAGE   := cartesi/machine-emulator-tools:$(VERSION)
+TOOLS_ROOTFS  := rootfs-tools-v$(VERSION).ext2
+
+IMAGE_KERNEL_VERSION ?= v0.19.0
+LINUX_VERSION ?= 6.5.9-ctsi-1
+LINUX_HEADERS_URLPATH := https://github.com/cartesi/image-kernel/releases/download/${IMAGE_KERNEL_VERSION}/linux-libc-dev-${LINUX_VERSION}-${IMAGE_KERNEL_VERSION}.deb
+
+all: fs
+
+build: control
+	@if ! (docker image inspect "$(TOOLS_IMAGE)" >/dev/null 2>&1) || [[ "$(force)" == "true" ]]; then \
+		docker buildx build --platform=linux/riscv64 --load \
+			--build-arg TOOLS_DEB=$(TOOLS_DEB) \
+			--build-arg IMAGE_KERNEL_VERSION=$(IMAGE_KERNEL_VERSION) \
+			--build-arg LINUX_VERSION=$(LINUX_VERSION) \
+			--build-arg LINUX_HEADERS_URLPATH=$(LINUX_HEADERS_URLPATH) \
+			-t $(TOOLS_IMAGE) \
+			-f Dockerfile \
+			. ; \
+	fi
+	@$(MAKE) copy
 
 copy:
-	ID=`docker create --platform=linux/riscv64 $(MACHINE_EMULATOR_TOOLS_IMAGE)` && \
-	   docker cp $$ID:/opt/cartesi/$(MACHINE_EMULATOR_TOOLS_TAR_GZ) . && \
-	   docker cp $$ID:/opt/cartesi/$(MACHINE_EMULATOR_TOOLS_DEB)    . && \
+	@ID=`docker create --platform=linux/riscv64 $(TOOLS_IMAGE)` && \
+	   docker cp $$ID:/opt/cartesi/$(TOOLS_DEB)    . && \
 	   docker rm $$ID
 
-deb:
-	mkdir -p $(DESTDIR)/DEBIAN
-	cat tools/template/control.template | sed 's|ARG_VERSION|$(VERSION)|g' > $(DESTDIR)/DEBIAN/control
-	cp -R $(STAGING_BASE)* $(DESTDIR)$(PREFIX)
-	dpkg-deb -Zxz --root-owner-group --build $(DESTDIR) $(MACHINE_EMULATOR_TOOLS_DEB)
-
-$(LINUX_SOURCES_FILEPATH):
-	T=`mktemp` && \
-	    mkdir -p `dirname $@` &&  \
-	    wget $(LINUX_SOURCES_URLPATH) -O $$T && \
-	    mv $$T $@ || (rm $$T && false)
+$(TOOLS_DEB) deb: build
```

I find this clearer.
```suggestion
deb: $(TOOLS_DEB)
$(TOOLS_DEB): build
```